### ISSUE TITLE
Unclear how to run preprocessor

### ISSUE THREAD

**Issue by noelwelsh** [rust-lang-nursery/mdBook#805]
I've created a [preprocessor](https://github.com/inner-product/mdbook-scala-wrapper) (using the 0.2.1 API, not the 0.2.2 alpha API, which seems to have some changes). It compiles ok, but I'm not sure how to get `mdbook` to actually run it. In my `book.toml` I have

```
[build]
preprocess = ["links", "index", "scala-wrapper"]
```

and `mdbook-scala-wrapper` is on my path, but I get 

```
[ERROR] (mdbook::utils): Error: "scala-wrapper" is not a recognised preprocessor
```

(In general the `mdbook` error messages could be much more informative, such as stating where it's looking for `SUMMARY.md` when it can't find it.)

**Comment by noelwelsh** [comment]
Ok, looking at the code it seems I can't add my own preprocessors to 0.2.1, which is annoying. I'll have to switch to 0.2.2-alpha and rewrite my preprocessor to use that API. I wonder why you bothered to document them in this case?

**Comment by Michael-F-Bryan** [comment]
@noelwelsh sorry about the poor user experience. We've only recently made the preprocessor system public so in `0.2.1` there wasn't really a way for users to inject their own `Preprocessor` implementation without creating a `mdbook` wrapper program (e.g. [mdbook-mermaid]).

I finally found some spare time, so I've made the final documentation changes we were waiting on and published `0.2.2` to crates.io.

> I wonder why you bothered to document them in this case?

I think the confusion is because the our CI system will automatically generate the user guide and push it to GitHub Pages every time there's a commit to `master`. So preprocessor docs had been merged into `master` and were viewable but not yet usable because `0.2.2` wasn't released until an hour ago.

[mdbook-mermaid]: https://github.com/badboy/mdbook-mermaid
